packages/php-table-backend-utils/src/Schema/Snowflake/SnowflakeSchemaReflection.php

```diff
@@ -68,10 +69,40 @@ public function getDefinitions(): array
             SnowflakeQuote::quote($this->schemaName),
         );
 
+        // Snowflake maps in DESC TABLE few data-type aliases to their basic types
+        // but in INFORMATION_SCHEMA.COLUMNS table keep data-type aliases
+        // here is implemented same mapping as DESC TABLE uses
         $columnsQuery = sprintf(
-            'SELECT TABLE_NAME, '.
-            'COLUMN_NAME AS "name", DATA_TYPE AS "type", COLUMN_DEFAULT AS "default", IS_NULLABLE AS "null?" '.
-            'FROM INFORMATION_SCHEMA.COLUMNS WHERE table_schema = %s ORDER BY TABLE_NAME;',
+            <<<SQL
+SELECT 
+    TABLE_NAME,
+    COLUMN_NAME AS "name",
+    CASE 
+        -- Map string types to VARCHAR
+        WHEN DATA_TYPE IN ('CHAR', 'VARCHAR', 'STRING', 'TEXT') THEN 
+            'VARCHAR(' || COALESCE(CHARACTER_MAXIMUM_LENGTH::STRING, '16777216') || ')'
+        
+        -- Map numeric types to NUMBER
+        WHEN DATA_TYPE IN ('NUMBER', 'DECIMAL', 'NUMERIC') THEN 
+            'NUMBER(' || COALESCE(NUMERIC_PRECISION::STRING, '') || ',' || COALESCE(NUMERIC_SCALE::STRING, '0') || ')'
+        
+        -- Map date and time types to their respective names - DATE is mapped directly to DATE
+        WHEN DATA_TYPE IN ('DATETIME', 'TIME', 'TIMESTAMP', 'TIMESTAMP_LTZ', 'TIMESTAMP_TZ', 'TIMESTAMP_NTZ') THEN 
+            DATA_TYPE || '(' || COALESCE(DATETIME_PRECISION::STRING, '') || ')'
+        
+        -- Map binary and varbinary - Snowflake don't have length for binary in INFORMATION_SCHEMA - handled in code
+        WHEN DATA_TYPE IN ('BINARY', 'VARBINARY') THEN
+            'BINARY' 
+ 
+        -- Default case for all other types as they are mapped by DESC TABLE direcly to themself
```

Comment:
premyslim jeste o dalsich datatypech (napr FLOAT...), ale asi to nechme tak

Reply:
pravda, mapují se všechny na `FLOAT`: https://docs.snowflake.com/en/sql-reference/data-types-numeric#examples-of-floating-point-data-types-in-a-table

To tam ještě přidám

Reply:
ne, to je prece v poradku, je to shodne s DESC table -> bude to float 
![image](https://github.com/user-attachments/assets/bb307b4d-0948-413f-8b3b-d18ec9ac6fdf)

![image](https://github.com/user-attachments/assets/c87cf69b-ecf8-4038-a4c5-07e12e5a84b7)

Reply:
ta myslenka byla v tom, ze jsi identifikoval nejake datatypy, ktere v te query resis. Ale neumime rict, jestli se vsechny ostatni pak chovaji "spravne" - tzn muzou defaultovat na to co o sobe opradvu rikaji

Reply:
Jo máš pravdu, float je OK.

Asi by teoreticky mohl bejt problém s novým data typem nebo pokud to snowflake změni.

Rád bych pak napsal test, který to bude porovnávat, ale asi až to celé doklepnu tak jako další udržovací task.

Reply:
jop -> https://keboolaglobal.slack.com/archives/C0553EM2RGX/p1737721669776799?thread_ts=1737718977.471229&cid=C0553EM2RGX

takze 👍